src/hooks/search.ts

```diff
@@ -20,22 +18,47 @@ export function useSearch(collection: Ref<CollectionMeta | null>, defaultCategor
       return collection.value.icons
   })
 
-  const fuse = computed(() => {
-    const icons = iconSource.value.map(icon => ({ icon }))
-    return markRaw(new Fuse(icons, {
-      includeScore: false,
-      keys: ['icon'],
+  const fzf = computed(() => {
+    return markRaw(new AsyncFzf(iconSource.value, {
+      casing: 'case-insensitive',
+      match: asyncExtendedMatch,
     }))
   })
 
-  const icons = computed(() => {
-    const searchString = throttledSearch.value.trim().toLowerCase()
-    if (!searchString)
-      return iconSource.value
-    else if (isAll.value) // disable fuse when search in all collections
-      return iconSource.value.filter(i => i.includes(searchString))
-    else
-      return fuse.value.search(searchString).map(i => i.item.icon)
+  const fzfFast = computed(() => {
+    return markRaw(new AsyncFzf(iconSource.value, {
+      casing: 'case-insensitive',
+      // v1 is faster
+      // https://fzf.netlify.app/docs/latest#async-finder-considering-other-options-first
+      fuzzy: 'v1',
+    }))
+  })
+
+  const icons = ref<string[]>([])
+
+  watchEffect(async () => {
```

Comment:
To avoid the pitfall of async context in the future, we'd better use a sync function here and use `.then` and `cache` for searching.